service/history/execution/mutable_state_builder.go

```diff
@@ -2132,6 +2135,28 @@ func (e *mutableStateBuilder) AddActivityTaskScheduledEvent(
 		return nil, nil, nil, false, false, e.createCallerError(opTag)
 	}
 
+	pendingActivitiesCount := len(e.pendingActivityInfoIDs)
+
+	if pendingActivitiesCount >= e.config.PendingActivitiesCountLimitError() {
+		e.logger.Error("Pending activity count exceeds error limit",
```

Comment:
definitely something we need metrics on, not just logs - this'll need automated alerts.